contrib/kube-prometheus/jsonnet/kube-prometheus/kube-prometheus-thanos.libsonnet

```diff
@@ -138,5 +138,44 @@ local servicePort = k.core.v1.service.mixin.spec.portsType;
       statefulSet.mixin.spec.template.spec.withVolumes([
         volume.fromEmptyDir('data'),
       ]),
+
+    thanosCompactorStatefulset:
+      local statefulSet = k.apps.v1beta2.statefulSet;
+      local volume = statefulSet.mixin.spec.template.spec.volumesType;
+      local container = statefulSet.mixin.spec.template.spec.containersType;
+      local containerEnv = container.envType;
+      local containerVolumeMount = container.volumeMountsType;
+
+      local labels = { app: 'thanos', 'thanos-peers': 'true' };
```

Comment:
It is not thanos-peer. It does not care about any gossip/dns/fileSD configuration. It does not participate in any Thanos inter communication and I guess that's where thanos-peers label is used?

Reply:
You're right, as this was just a hacky way to enable the metrics from that endpoint yesterday. I need to add another ServiceMonitor to match that separately, I guess. Thankfully this is still a WIP PR right now :blush: